tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.cc

```diff
@@ -207,9 +207,16 @@ void* GpuCudaMallocAsyncAllocator::AllocateRaw(size_t alignment,
   }
   se::cuda::ScopedActivateExecutorContext scoped_activation{stream_exec_};
   void* ptr = nullptr;
-  if (auto result =
-          cuMemAllocFromPoolAsync(reinterpret_cast<CUdeviceptr*>(&ptr),
-                                  num_bytes, pool_, cuda_stream_)) {
+  auto result = cuMemAllocFromPoolAsync(reinterpret_cast<CUdeviceptr*>(&ptr),
+                                        num_bytes, pool_, cuda_stream_);
+  if (result == CUDA_ERROR_OUT_OF_MEMORY) {
+    // This can fix some OOM when memory is tight.
```

Comment:
Can you please add some more detail on the situations in which this helps?

Reply:
done.